tests/test_contacts.py

```diff
@@ -16,29 +16,29 @@
 
 @pytest.mark.unit
 @pytest.mark.parametrize(
-    ("assort_by", "expected"),
+    "states, group_code_name, expected",
     [
         (
-            ["age_group", "region"],
-            [[9, 12], [7, 10, 13], [8, 11, 14], [0, 3, 6], [1, 4], [2, 5]],
+            pd.DataFrame({"a": [1] * 7 + [0] * 8}),
+            ["a"],
+            [list(range(7, 15)), list(range(7))],
+        ),
+        (
+            pd.DataFrame({"a": pd.Series([0, 1, 2, 3, 0, 1, 2, 3]).astype("category")}),
+            ["a"],
+            [[0, 4], [1, 5], [2, 6], [3, 7]],
+        ),
+        (
+            pd.DataFrame({"a": pd.Series([0, 1, 2, 3, 0, 1, 2, -1])}),
+            ["a"],
+            [[7], [0, 4], [1, 5], [2, 6], [3]],
```

Comment:
I think the last test case is incorrect. The `-1` should just be omitted. The correct output would be `[[0, 4], [1, 5], [2, 6], [3]]`. We never look up negative group codes anyways (see [here](https://github.com/covid-19-impact-lab/sid/blob/fae23c92df6a442a4c441f366eb8a260ed833c52/src/sid/contacts.py#L267)) and all other group codes would be off by one if we leave it in!